pkg/agent/core/ngt/handler/grpc/handler_test.go

```diff
@@ -1188,57 +1209,1260 @@ func Test_server_Insert(t *testing.T) {
-		           name: "",
-		           ip: "",
-		           ngt: nil,
-		           eg: nil,
-		           streamConcurrency: 0,
-		       },
-		       want: want{},
-		       checkFunc: defaultCheckFunc,
-		   },
-		*/
 
-		// TODO test cases
-		/*
-		   func() test {
-		       return test {
-		           name: "test_case_2",
-		           args: args {
-		           ctx: nil,
-		           req: nil,
-		           },
-		           fields: fields {
-		           name: "",
-		           ip: "",
-		           ngt: nil,
-		           eg: nil,
-		           streamConcurrency: 0,
-		           },
-		           want: want{},
-		           checkFunc: defaultCheckFunc,
-		       }
-		   }(),
-		*/
+	/*
+		- Equivalence Class Testing
+			- uint8, float32
+				- case 1.1: Insert vector success (vector type is uint8)
+				- case 1.2: Insert vector success (vector type is float32)
+				- case 2.1: Insert vector with different dimension (vector type is uint8)
+				- case 2.2: Insert vector with different dimension (vector type is float32)
+
+		- Boundary Value Testing
+			- uint8, float32
+				- case 1.1: Insert vector with 0 value success (vector type is uint8)
+				- case 1.1: Insert vector with 0 value success (vector type is float32)
+				- case 2.1: Insert vector with min value success (vector type is uint8)
+				- case 2.2: Insert vector with min value success (vector type is float32)
+				- case 3.1: Insert vector with max value success (vector type is uint8)
+				- case 3.2: Insert vector with max value success (vector type is float32)
+				- case 4.1: Insert with empty UUID fail (vector type is uint8)
+				- case 4.2: Insert with empty UUID fail (vector type is float32)
+
+			- float32
+				- case 5: Insert vector with NaN value fail (vector type is float32)
+
+			- case 6: Insert nil insert request fail
+				* IncompatibleDimensionSize error will be returned.
+			- case 7: Insert nil vector fail
+				* IncompatibleDimensionSize error will be returned.
+			- case 8: Insert empty insert vector fail
+				* IncompatibleDimensionSize error will be returned.
+
+		- Decision Table Testing
+			- duplicated ID, duplicated vector, duplicated ID & vector
+				- case 1.1: Insert duplicated request fail when SkipStrictExistCheck is off (duplicated ID)
+					* AlreadyExists error will be returned.
+				- case 1.2: Insert duplicated request success when SkipStrictExistCheck is off (duplicated vector)
+				- case 1.3: Insert duplicated request fail when SkipStrictExistCheck is off (duplicated ID & vector)
+				- case 2.1: Insert duplicated request success when SkipStrictExistCheck is on (duplicated ID)
+				- case 2.2: Insert duplicated request success when SkipStrictExistCheck is on (duplicated vector)
+				- case 2.3: Insert duplicated request success when SkipStrictExistCheck is on (duplicated ID & vector)
+	*/
+	tests := []test{
+		// Equivalence Class Testing
+		func() test {
+			name := "vald-agent-ngt-1"
+			id := "uuid1"
+			ip := "127.0.0.1"
```

Comment:
We should get the IP address from the environment.

Reply:
"127.0.0.1" is the local host IP address, I think it is ok in this test case.
What do you think?

Reply:
btw, you have to define it as a constant value if you use it for some test cases.

Reply:
Sorry, I misunderstood, but I found the LoadLocalIP function in internal/net package.
I think we should use it.

Reply:
updated :) thank you for your comment